<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Fit linear mixed model for differential expression and preform hypothesis test on fixed effects as specified in the contrast matrix L"><title>Differential expression with linear mixed model — dream • variancePartition</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Differential expression with linear mixed model — dream"><meta property="og:description" content="Fit linear mixed model for differential expression and preform hypothesis test on fixed effects as specified in the contrast matrix L"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary" data-bs-theme="dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">variancePartition</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.33.15</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/variancePartition.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/dream.html">dream analysis</a>
    <a class="dropdown-item" href="../articles/rnd_effects.html">Theory and practice of random effects</a>
    <a class="dropdown-item" href="../articles/mvtests.html">Multivariate tests</a>
    <a class="dropdown-item" href="../articles/additional_visualization.html">Additional visualizations of variance structure</a>
    <a class="dropdown-item" href="../articles/errors.html">Error handling</a>
    <a class="dropdown-item" href="../articles/FAQ.html">Frequently asked questions</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/DiseaseNeuroGenomics/variancePartition/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Differential expression with linear mixed model</h1>
      <small class="dont-index">Source: <a href="https://github.com/DiseaseNeuroGenomics/variancePartition/blob/HEAD/R/dream.R" class="external-link"><code>R/dream.R</code></a></small>
      <div class="d-none name"><code>dream-method.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Fit linear mixed model for differential expression and preform hypothesis test on fixed effects as specified in the contrast matrix <code>L</code></p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">dream</span><span class="op">(</span></span>
<span>  <span class="va">exprObj</span>,</span>
<span>  <span class="va">formula</span>,</span>
<span>  <span class="va">data</span>,</span>
<span>  <span class="va">L</span>,</span>
<span>  ddf <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"adaptive"</span>, <span class="st">"Satterthwaite"</span>, <span class="st">"Kenward-Roger"</span><span class="op">)</span>,</span>
<span>  useWeights <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  control <span class="op">=</span> <span class="va">vpcontrol</span>,</span>
<span>  hideErrorsInBackend <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  REML <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  BPPARAM <span class="op">=</span> <span class="fu">SerialParam</span><span class="op">(</span><span class="op">)</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>exprObj</dt>
<dd><p>matrix of expression data (g genes x n samples), or <code>ExpressionSet</code>, or <code>EList</code> returned by voom() from the limma package</p></dd>


<dt>formula</dt>
<dd><p>specifies variables for the linear (mixed) model.  Must only specify covariates, since the rows of exprObj are automatically used as a response. e.g.: <code>~ a + b + (1|c)</code>  Formulas with only fixed effects also work, and <code><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit()</a></code> followed by <code><a href="https://rdrr.io/pkg/limma/man/contrasts.fit.html" class="external-link">contrasts.fit()</a></code> are run.</p></dd>


<dt>data</dt>
<dd><p>data.frame with columns corresponding to formula</p></dd>


<dt>L</dt>
<dd><p>contrast matrix specifying a linear combination of fixed effects to test</p></dd>


<dt>ddf</dt>
<dd><p>Specifiy "Satterthwaite" or "Kenward-Roger" method to estimate effective degress of freedom for hypothesis testing in the linear mixed model.  Note that Kenward-Roger is more accurate, but is *much* slower.  Satterthwaite is a good enough approximation for most datasets. "adaptive" (Default) uses KR for &lt;= 20 samples.</p></dd>


<dt>useWeights</dt>
<dd><p>if TRUE, analysis uses heteroskedastic error estimates from <code><a href="https://rdrr.io/pkg/limma/man/voom.html" class="external-link">voom()</a></code>.  Value is ignored unless exprObj is an <code>EList()</code> from <code><a href="https://rdrr.io/pkg/limma/man/voom.html" class="external-link">voom()</a></code> or <code>weightsMatrix</code> is specified</p></dd>


<dt>control</dt>
<dd><p>control settings for <code><a href="https://rdrr.io/pkg/lme4/man/lmer.html" class="external-link">lmer()</a></code></p></dd>


<dt>hideErrorsInBackend</dt>
<dd><p>default FALSE.  If TRUE, hide errors in <code>attr(.,"errors")</code> and <code>attr(.,"error.initial")</code></p></dd>


<dt>REML</dt>
<dd><p>use restricted maximum likelihood to fit linear mixed model. default is TRUE.  See Details.</p></dd>


<dt>BPPARAM</dt>
<dd><p>parameters for parallel evaluation</p></dd>


<dt>...</dt>
<dd><p>Additional arguments for <code><a href="https://rdrr.io/pkg/lme4/man/lmer.html" class="external-link">lmer()</a></code> or <code><a href="https://rdrr.io/r/stats/lm.html" class="external-link">lm()</a></code></p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>MArrayLM2 object (just like MArrayLM from limma), and the directly estimated p-value (without eBayes)</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>A linear (mixed) model is fit for each gene in exprObj, using formula to specify variables in the regression (Hoffman and Roussos, 2021).  If categorical variables are modeled as random effects (as is recommended), then a linear mixed model us used.  For example if formula is <code>~ a + b + (1|c)</code>, then the model is</p>
<p><code>fit &lt;- lmer( exprObj[j,] ~ a + b + (1|c), data=data)</code></p>
<p><code>useWeights=TRUE</code> causes <code>weightsMatrix[j,]</code> to be included as weights in the regression model.</p>
<p>Note: Fitting the model for 20,000 genes can be computationally intensive.  To accelerate computation, models can be fit in parallel using <code>BiocParallel</code> to run code in parallel.  Parallel processing must be enabled before calling this function.  See below.</p>
<p>The regression model is fit for each gene separately. Samples with missing values in either gene expression or metadata are omitted by the underlying call to lmer.</p>
<p>Hypothesis tests and degrees of freedom are producted by <code>lmerTest</code> and <code>pbkrtest</code> pacakges</p>
<p>While <code>REML=TRUE</code> is required by <code>lmerTest</code> when ddf='Kenward-Roger', ddf='Satterthwaite' can be used with <code>REML</code> as <code>TRUE</code> or <code>FALSE</code>.  Since the Kenward-Roger method gave the best power with an accurate control of false positive rate in our simulations, and since the Satterthwaite method with REML=TRUE gives p-values that are slightly closer to the Kenward-Roger p-values, <code>REML=TRUE</code> is the default.  See Vignette "3) Theory and practice of random effects and REML"</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Hoffman GE, Roussos P (2021).
“dream: Powerful differential expression analysis for repeated measures designs.”
<em>Bioinformatics</em>, <b>37</b>(2), 192--201.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># library(variancePartition)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># load simulated data:</span></span></span>
<span class="r-in"><span><span class="co"># geneExpr: matrix of *normalized* gene expression values</span></span></span>
<span class="r-in"><span><span class="co"># info: information/metadata about each sample</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">varPartData</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">form</span> <span class="op">&lt;-</span> <span class="op">~</span> <span class="va">Batch</span> <span class="op">+</span> <span class="op">(</span><span class="fl">1</span> <span class="op">|</span> <span class="va">Individual</span><span class="op">)</span> <span class="op">+</span> <span class="op">(</span><span class="fl">1</span> <span class="op">|</span> <span class="va">Tissue</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Fit linear mixed model for each gene</span></span></span>
<span class="r-in"><span><span class="co"># run on just 10 genes for time</span></span></span>
<span class="r-in"><span><span class="co"># NOTE: dream() runs on *normalized* data</span></span></span>
<span class="r-in"><span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu">dream</span><span class="op">(</span><span class="va">geneExpr</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>, <span class="va">form</span>, <span class="va">info</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># view top genes</span></span></span>
<span class="r-in"><span><span class="fu"><a href="toptable-method.html">topTable</a></span><span class="op">(</span><span class="va">fit</span>, coef <span class="op">=</span> <span class="st">"Batch2"</span>, number <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            logFC    AveExpr         t   P.Value adj.P.Val         B     z.std</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene6 -0.6704969  -3.155479 -1.651195 0.1007713 0.6907345 -4.539538 -1.641126</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene2 -0.4942074  -1.128161 -1.340326 0.1821535 0.6907345 -4.569990 -1.334154</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene1 -0.4609243 -10.466455 -1.266674 0.2072203 0.6907345 -4.576299 -1.261246</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># get contrast matrix testing if the coefficient for Batch3 is</span></span></span>
<span class="r-in"><span><span class="co"># different from coefficient for Batch2</span></span></span>
<span class="r-in"><span><span class="co"># Name this comparison as 'compare_3_2'</span></span></span>
<span class="r-in"><span><span class="co"># The variable of interest must be a fixed effect</span></span></span>
<span class="r-in"><span><span class="va">L</span> <span class="op">&lt;-</span> <span class="fu"><a href="makeContrastsDream.html">makeContrastsDream</a></span><span class="op">(</span><span class="va">form</span>, <span class="va">info</span>, contrasts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>compare_3_2 <span class="op">=</span> <span class="st">"Batch3 - Batch2"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># plot contrasts</span></span></span>
<span class="r-in"><span><span class="fu"><a href="plotContrasts.html">plotContrasts</a></span><span class="op">(</span><span class="va">L</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="dream-method-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Fit linear mixed model for each gene</span></span></span>
<span class="r-in"><span><span class="co"># run on just 10 genes for time</span></span></span>
<span class="r-in"><span><span class="va">fit2</span> <span class="op">&lt;-</span> <span class="fu">dream</span><span class="op">(</span><span class="va">geneExpr</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>, <span class="va">form</span>, <span class="va">info</span>, <span class="va">L</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">fit2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fit2</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># view top genes for this contrast</span></span></span>
<span class="r-in"><span><span class="fu"><a href="toptable-method.html">topTable</a></span><span class="op">(</span><span class="va">fit2</span>, coef <span class="op">=</span> <span class="st">"compare_3_2"</span>, number <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            logFC    AveExpr          t   P.Value adj.P.Val         B      z.std</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene7  0.4655178 -4.3799381  1.4018138 0.1630310 0.8184637 -4.557141  1.3949498</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene6 -0.3642792 -3.1554790 -1.0121895 0.3130692 0.8184637 -4.595941 -1.0088028</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene3 -0.2545920  0.1702122 -0.8117287 0.4182311 0.8184637 -4.611148 -0.8094939</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Parallel processing using multiple cores with reduced memory usage</span></span></span>
<span class="r-in"><span><span class="va">param</span> <span class="op">&lt;-</span> <span class="fu">SnowParam</span><span class="op">(</span><span class="fl">4</span>, <span class="st">"SOCK"</span>, progressbar <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">fit3</span> <span class="op">&lt;-</span> <span class="fu">dream</span><span class="op">(</span><span class="va">geneExpr</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>, <span class="va">form</span>, <span class="va">info</span>, <span class="va">L</span>, BPPARAM <span class="op">=</span> <span class="va">param</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> iteration: </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span><span class="va">fit3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fit3</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Fit fixed effect model for each gene</span></span></span>
<span class="r-in"><span><span class="co"># Use lmFit in the backend</span></span></span>
<span class="r-in"><span><span class="va">form</span> <span class="op">&lt;-</span> <span class="op">~</span><span class="va">Batch</span></span></span>
<span class="r-in"><span><span class="va">fit4</span> <span class="op">&lt;-</span> <span class="fu">dream</span><span class="op">(</span><span class="va">geneExpr</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>, <span class="va">form</span>, <span class="va">info</span>, <span class="va">L</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">fit4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fit4</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># view top genes</span></span></span>
<span class="r-in"><span><span class="fu"><a href="toptable-method.html">topTable</a></span><span class="op">(</span><span class="va">fit4</span>, coef <span class="op">=</span> <span class="st">"compare_3_2"</span>, number <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           logFC    AveExpr          t    P.Value adj.P.Val         B</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene8 -1.835677  0.9171386 -1.8703068 0.06312128 0.6312128 -4.581611</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene4 -1.016974 -4.5359748 -1.1783589 0.24026308 0.9496566 -4.593019</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene9 -0.711481 -2.3079042 -0.6924022 0.48960809 0.9496566 -4.598021</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Compute residuals using dream</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/residuals.html" class="external-link">residuals</a></span><span class="op">(</span><span class="va">fit4</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             s1         s2         s3        s4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene1 2.196588 -7.2102826  1.3618545 0.8055865</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene2 1.270341  0.6885371 -0.8045642 2.0594469</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene3 1.939465  0.8003329 -1.7723606 3.0120196</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene4 2.766271 -2.4954702 -3.5198043 5.2286010</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="http://gabrielhoffman.github.io" class="external-link">Gabriel Hoffman</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer></div>

  

  

  </body></html>

