<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Function reference â€¢ variancePartition</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary" data-bs-theme="dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">variancePartition</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.35.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/variancePartition.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/dream.html">dream analysis</a>
    <a class="dropdown-item" href="../articles/rnd_effects.html">Theory and practice of random effects</a>
    <a class="dropdown-item" href="../articles/mvtests.html">Multivariate tests</a>
    <a class="dropdown-item" href="../articles/additional_visualization.html">Additional visualizations of variance structure</a>
    <a class="dropdown-item" href="../articles/errors.html">Error handling</a>
    <a class="dropdown-item" href="../articles/FAQ.html">Frequently asked questions</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/DiseaseNeuroGenomics/variancePartition/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Function reference</h1>
    </div>

    <div class="section level2">
      <h2 id="all-functions">All functions<a class="anchor" aria-label="anchor" href="#all-functions"></a></h2>
      
      

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="BIC.MArrayLM.html">BIC(<i>&lt;MArrayLM&gt;</i>)</a></code> 
        </dt>
        <dd>BIC from model fit</dd>
      </dl><dl><dt>
          
          <code><a href="BIC.MArrayLM2.html">BIC(<i>&lt;MArrayLM2&gt;</i>)</a></code> 
        </dt>
        <dd>BIC from model fit</dd>
      </dl><dl><dt>
          
          <code><a href="ESS-method.html">ESS()</a></code> 
        </dt>
        <dd>Effective sample size</dd>
      </dl><dl><dt>
          
          <code><a href="MArrayLM2-class.html">MArrayLM2-class</a></code> 
        </dt>
        <dd>Class MArrayLM2</dd>
      </dl><dl><dt>
          
          <code><a href="VarParCIList-class.html">VarParCIList-class</a></code> 
        </dt>
        <dd>Class VarParCIList</dd>
      </dl><dl><dt>
          
          <code><a href="VarParFitList-class.html">VarParFitList-class</a></code> 
        </dt>
        <dd>Class VarParFitList</dd>
      </dl><dl><dt>
          
          <code><a href="applyQualityWeights.html">applyQualityWeights()</a></code> 
        </dt>
        <dd>Apply pre-specified sample weights</dd>
      </dl><dl><dt>
          
          <code><a href="as.data.frame.varPartResults.html">as.data.frame(<i>&lt;varPartResults&gt;</i>)</a></code> 
        </dt>
        <dd>Convert to data.frame</dd>
      </dl><dl><dt>
          
          <code><a href="as.matrix-varPartResults-method.html">as.matrix(<i>&lt;varPartResults&gt;</i>)</a></code> 
        </dt>
        <dd>Convert to matrix</dd>
      </dl><dl><dt>
          
          <code><a href="augmentPriorCount.html">augmentPriorCount()</a></code> 
        </dt>
        <dd>Augment observed read counts with prior count</dd>
      </dl><dl><dt>
          
          <code><a href="calcVarPart-method.html">calcVarPart()</a></code> 
        </dt>
        <dd>Compute variance statistics</dd>
      </dl><dl><dt>
          
          <code><a href="canCorPairs.html">canCorPairs()</a></code> 
        </dt>
        <dd>canCorPairs</dd>
      </dl><dl><dt>
          
          <code><a href="classifyTestsF-MArrayLM2-method.html">classifyTestsF(<i>&lt;MArrayLM2&gt;</i>)</a></code> 
        </dt>
        <dd>Multiple Testing Genewise Across Contrasts</dd>
      </dl><dl><dt>
          
          <code><a href="classifyTestsF.html">classifyTestsF()</a></code> 
        </dt>
        <dd>Multiple Testing Genewise Across Contrasts</dd>
      </dl><dl><dt>
          
          <code><a href="colinearityScore.html">colinearityScore()</a></code> 
        </dt>
        <dd>Collinearity score</dd>
      </dl><dl><dt>
          
          <code><a href="deviation-method.html">deviation()</a></code> 
        </dt>
        <dd>Deviation from expectation for each observation</dd>
      </dl><dl><dt>
          
          <code><a href="diffVar-method.html">diffVar()</a></code> 
        </dt>
        <dd>Test differential variance</dd>
      </dl><dl><dt>
          
          <code><a href="dream-method.html">dream()</a></code> 
        </dt>
        <dd>Differential expression with linear mixed model</dd>
      </dl><dl><dt>
          
          <code><a href="dscchisq.html">dscchisq()</a></code> 
        </dt>
        <dd>Scaled chi-square</dd>
      </dl><dl><dt>
          
          <code><a href="eBayes-method.html">eBayes(<i>&lt;MArrayLM2&gt;</i>)</a></code> 
        </dt>
        <dd>eBayes for MArrayLM2</dd>
      </dl><dl><dt>
          
          <code><a href="extractVarPart.html">extractVarPart()</a></code> 
        </dt>
        <dd>Extract variance statistics</dd>
      </dl><dl><dt>
          
          <code><a href="fitExtractVarPartModel-method.html">fitExtractVarPartModel()</a></code> 
        </dt>
        <dd>Fit linear (mixed) model, report variance fractions</dd>
      </dl><dl><dt>
          
          <code><a href="fitVarPartModel-method.html">fitVarPartModel()</a></code> 
        </dt>
        <dd>Fit linear (mixed) model</dd>
      </dl><dl><dt>
          
          <code><a href="getContrast-method.html">getContrast()</a></code> 
        </dt>
        <dd>Extract contrast matrix for linear mixed model</dd>
      </dl><dl><dt>
          
          <code><a href="getTreat-method.html">getTreat()</a></code> 
        </dt>
        <dd>Test if coefficient is different from a specified value</dd>
      </dl><dl><dt>
          
          <code><a href="get_prediction-method.html">get_prediction()</a></code> 
        </dt>
        <dd>Compute predicted value of formula for linear (mixed) model</dd>
      </dl><dl><dt>
          
          <code><a href="ggColorHue.html">ggColorHue()</a></code> 
        </dt>
        <dd>Default colors for ggplot</dd>
      </dl><dl><dt>
          
          <code><a href="hatvalues-method.html">hatvalues(<i>&lt;MArrayLM&gt;</i>)</a></code> <code><a href="hatvalues-method.html">hatvalues(<i>&lt;MArrayLM2&gt;</i>)</a></code> 
        </dt>
        <dd>Compute hatvalues</dd>
      </dl><dl><dt>
          
          <code><a href="isRunableFormula.html">isRunableFormula()</a></code> 
        </dt>
        <dd>Test if formula is full rank on this dataset</dd>
      </dl><dl><dt>
          
          <code><a href="logLik.MArrayLM.html">logLik(<i>&lt;MArrayLM&gt;</i>)</a></code> 
        </dt>
        <dd>Log-likelihood from model fit</dd>
      </dl><dl><dt>
          
          <code><a href="logLik.MArrayLM2.html">logLik(<i>&lt;MArrayLM2&gt;</i>)</a></code> 
        </dt>
        <dd>Log-likelihood from model fit</dd>
      </dl><dl><dt>
          
          <code><a href="makeContrastsDream.html">makeContrastsDream()</a></code> 
        </dt>
        <dd>Construct Matrix of Custom Contrasts</dd>
      </dl><dl><dt>
          
          <code><a href="mvTest-method.html">mvTest()</a></code> 
        </dt>
        <dd>Multivariate tests on results from <code><a href="../reference/dream-method.html">dream()</a></code></dd>
      </dl><dl><dt>
          
          <code><a href="mvTest_input-class.html">mvTest_input-class</a></code> 
        </dt>
        <dd>Class mvTest_input</dd>
      </dl><dl><dt>
          
          <code><a href="plotCompareP-method.html">plotCompareP()</a></code> 
        </dt>
        <dd>Compare p-values from two analyses</dd>
      </dl><dl><dt>
          
          <code><a href="plotContrasts.html">plotContrasts()</a></code> 
        </dt>
        <dd>Plot representation of contrast matrix</dd>
      </dl><dl><dt>
          
          <code><a href="plotCorrMatrix.html">plotCorrMatrix()</a></code> 
        </dt>
        <dd>plotCorrMatrix</dd>
      </dl><dl><dt>
          
          <code><a href="plotCorrStructure.html">plotCorrStructure()</a></code> 
        </dt>
        <dd>plotCorrStructure</dd>
      </dl><dl><dt>
          
          <code><a href="plotPercentBars-method.html">plotPercentBars()</a></code> 
        </dt>
        <dd>Bar plot of gene fractions</dd>
      </dl><dl><dt>
          
          <code><a href="plotStratify.html">plotStratify()</a></code> 
        </dt>
        <dd>plotStratify</dd>
      </dl><dl><dt>
          
          <code><a href="plotStratifyBy.html">plotStratifyBy()</a></code> 
        </dt>
        <dd>plotStratifyBy</dd>
      </dl><dl><dt>
          
          <code><a href="plotVarPart-method.html">plotVarPart()</a></code> 
        </dt>
        <dd>Violin plot of variance fractions</dd>
      </dl><dl><dt>
          
          <code><a href="plotVarianceEstimates.html">plotVarianceEstimates()</a></code> 
        </dt>
        <dd>Plot Variance Estimates</dd>
      </dl><dl><dt>
          
          <code><a href="rdf.html">rdf()</a></code> 
        </dt>
        <dd>Residual degrees of freedom</dd>
      </dl><dl><dt>
          
          <code><a href="rdf.merMod.html">rdf.merMod()</a></code> 
        </dt>
        <dd>Approximate residual degrees of freedom</dd>
      </dl><dl><dt>
          
          <code><a href="rdf_from_matrices.html">rdf_from_matrices()</a></code> 
        </dt>
        <dd>Fast approximate residual degrees of freedom</dd>
      </dl><dl><dt>
          
          <code><a href="reOnly.html">reOnly()</a></code> 
        </dt>
        <dd>Adapted from lme4:::reOnly</dd>
      </dl><dl><dt>
          
          <code><a href="residuals-MArrayLM-method.html">residuals(<i>&lt;MArrayLM&gt;</i>)</a></code> 
        </dt>
        <dd>residuals for MArrayLM</dd>
      </dl><dl><dt>
          
          <code><a href="residuals-MArrayLM2-method.html">residuals(<i>&lt;MArrayLM2&gt;</i>)</a></code> 
        </dt>
        <dd>residuals for MArrayLM2</dd>
      </dl><dl><dt>
          
          <code><a href="residuals-VarParFitList-method.html">residuals(<i>&lt;VarParFitList&gt;</i>)</a></code> 
        </dt>
        <dd>Residuals from model fit</dd>
      </dl><dl><dt>
          
          <code><a href="residuals.MArrayLM2.html">residuals.MArrayLM2()</a></code> 
        </dt>
        <dd>Residuals for result of dream</dd>
      </dl><dl><dt>
          
          <code><a href="shrinkageMetric.html">shrinkageMetric()</a></code> 
        </dt>
        <dd>Shrinkage metric for eBayes</dd>
      </dl><dl><dt>
          
          <code><a href="sortCols-method.html">sortCols()</a></code> 
        </dt>
        <dd>Sort variance partition statistics</dd>
      </dl><dl><dt>
          
          <code><a href="toptable-method.html">topTable()</a></code> 
        </dt>
        <dd>Table of Top Genes from Linear Model Fit</dd>
      </dl><dl><dt>
          
          <code><a href="varParFrac-class.html">varParFrac-class</a></code> 
        </dt>
        <dd>Class varParFrac</dd>
      </dl><dl><dt>
          
          <code><a href="varPartConfInf.html">varPartConfInf()</a></code> 
        </dt>
        <dd>Linear mixed model confidence intervals</dd>
      </dl><dl><dt>
          
          <code><a href="varPartDEdata.html">varPartData</a></code> 
        </dt>
        <dd>A simulated dataset of gene counts</dd>
      </dl><dl><dt>
          
          <code><a href="varPartData.html">varPartData</a></code> 
        </dt>
        <dd>Simulation dataset for examples</dd>
      </dl><dl><dt>
          
          <code><a href="varPartResults-class.html">varPartResults-class</a></code> 
        </dt>
        <dd>Class varPartResults</dd>
      </dl><dl><dt>
          
          <code><a href="vcov-MArrayLM-method.html">vcov(<i>&lt;MArrayLM&gt;</i>)</a></code> 
        </dt>
        <dd>Co-variance matrix for <code><a href="../reference/dream-method.html">dream()</a></code> fit</dd>
      </dl><dl><dt>
          
          <code><a href="vcov-MArrayLM2-method.html">vcov(<i>&lt;MArrayLM2&gt;</i>)</a></code> 
        </dt>
        <dd>Co-variance matrix for <code><a href="../reference/dream-method.html">dream()</a></code> fit</dd>
      </dl><dl><dt>
          
          <code><a href="vcovSqrt-method.html">vcovSqrt()</a></code> 
        </dt>
        <dd>Sqrt of co-variance matrix for <code><a href="../reference/dream-method.html">dream()</a></code> fit</dd>
      </dl><dl><dt>
          
          <code><a href="voomWithDreamWeights.html">voomWithDreamWeights()</a></code> 
        </dt>
        <dd>Transform RNA-Seq Data Ready for Linear Mixed Modelling with <code><a href="../reference/dream-method.html">dream()</a></code></dd>
      </dl></div>
  </main></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="http://gabrielhoffman.github.io" class="external-link">Gabriel Hoffman</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer></div>

  

  

  </body></html>

