<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Construct the contrast matrix corresponding to specified contrasts of a set of parameters. Each specified set of contrast weights must sum to 1."><title>Construct Matrix of Custom Contrasts — makeContrastsDream • variancePartition</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Construct Matrix of Custom Contrasts — makeContrastsDream"><meta property="og:description" content="Construct the contrast matrix corresponding to specified contrasts of a set of parameters. Each specified set of contrast weights must sum to 1."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary" data-bs-theme="dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">variancePartition</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.33.15</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/variancePartition.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/dream.html">dream analysis</a>
    <a class="dropdown-item" href="../articles/rnd_effects.html">Theory and practice of random effects</a>
    <a class="dropdown-item" href="../articles/mvtests.html">Multivariate tests</a>
    <a class="dropdown-item" href="../articles/additional_visualization.html">Additional visualizations of variance structure</a>
    <a class="dropdown-item" href="../articles/errors.html">Error handling</a>
    <a class="dropdown-item" href="../articles/FAQ.html">Frequently asked questions</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/DiseaseNeuroGenomics/variancePartition/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Construct Matrix of Custom Contrasts</h1>
      <small class="dont-index">Source: <a href="https://github.com/DiseaseNeuroGenomics/variancePartition/blob/HEAD/R/contrasts.R" class="external-link"><code>R/contrasts.R</code></a></small>
      <div class="d-none name"><code>makeContrastsDream.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Construct the contrast matrix corresponding to specified contrasts of a set of parameters. Each specified set of contrast weights must sum to 1.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">makeContrastsDream</span><span class="op">(</span></span>
<span>  <span class="va">formula</span>,</span>
<span>  <span class="va">data</span>,</span>
<span>  <span class="va">...</span>,</span>
<span>  contrasts <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  suppressWarnings <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  nullOnError <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>formula</dt>
<dd><p>specifies variables for the linear (mixed) model.  Must only specify covariates, since the rows of exprObj are automatically used as a response. e.g.: <code>~ a + b + (1|c)</code>  Formulas with only fixed effects also work</p></dd>


<dt>data</dt>
<dd><p>data.frame with columns corresponding to formula</p></dd>


<dt>...</dt>
<dd><p>expressions, or character strings which can be parsed to expressions, specifying contrasts</p></dd>


<dt>contrasts</dt>
<dd><p>character vector specifying contrasts</p></dd>


<dt>suppressWarnings</dt>
<dd><p>(default FALSE). suppress warnings for univariate contrasts</p></dd>


<dt>nullOnError</dt>
<dd><p>(default FALSE). When a contrast entry is invalid, throw warning and return NULL for that contrast entry</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>matrix of linear contrasts between regression coefficients</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This function expresses contrasts between a set of parameters as a numeric matrix. The parameters are usually the coefficients from a linear (mixed) model fit, so the matrix specifies which comparisons between the coefficients are to be extracted from the fit. The output from this function is usually used as input to <code><a href="dream-method.html">dream()</a></code>.</p>
<p>This function creates a matrix storing the contrasts weights that are applied to each coefficient.</p>
<p>Consider a variable <code>v</code> with levels <code>c('A', 'B', 'C')</code>.  A contrast comparing <code>A</code> and <code>B</code> is <code>'vA - vB'</code> and tests whether the difference between these levels is different than zero. Coded for the 3 levels this has weights <code>c(1, -1, 0)</code>.  In order to compare <code>A</code> to the other levels, the contrast is <code>'vA - (vB + vC)/2'</code> so that <code>A</code> is compared to the average of the other two levels. This is encoded as <code>c(1, -0.5, -0.5)</code>.  This type of proper matching in testing multiple levels is enforced by ensuring that the contrast weights sum to 1. Based on standard regression theory only weighted sums of the estimated coefficients are supported.</p>
<p>This function is inspired by <code><a href="https://rdrr.io/pkg/limma/man/makeContrasts.html" class="external-link">limma::makeContrasts()</a></code> but is designed to be compatible with linear mixed models for <code><a href="dream-method.html">dream()</a></code></p>
<p>Names in ... and contrasts will be used as column names in the returned value.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="plotContrasts.html">plotContrasts()</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># load library</span></span></span>
<span class="r-in"><span><span class="co"># library(variancePartition)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Bioconductor/BiocParallel" class="external-link">BiocParallel</a></span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># load simulated data:</span></span></span>
<span class="r-in"><span><span class="co"># geneExpr: matrix of gene expression values</span></span></span>
<span class="r-in"><span><span class="co"># info: information/metadata about each sample</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">varPartData</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">form</span> <span class="op">&lt;-</span> <span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">Batch</span> <span class="op">+</span> <span class="op">(</span><span class="fl">1</span> <span class="op">|</span> <span class="va">Individual</span><span class="op">)</span> <span class="op">+</span> <span class="op">(</span><span class="fl">1</span> <span class="op">|</span> <span class="va">Tissue</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Define contrasts</span></span></span>
<span class="r-in"><span><span class="co"># Note that for each contrass, the weights sum to 1</span></span></span>
<span class="r-in"><span><span class="va">L</span> <span class="op">&lt;-</span> <span class="fu">makeContrastsDream</span><span class="op">(</span><span class="va">form</span>, <span class="va">info</span>, contrasts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>Batch1_vs_2 <span class="op">=</span> <span class="st">"Batch1 - Batch2"</span>, Batch3_vs_4 <span class="op">=</span> <span class="st">"Batch3 - Batch4"</span>, Batch1_vs_34 <span class="op">=</span> <span class="st">"Batch1 - (Batch3 + Batch4)/2"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># show contrasts matrix</span></span></span>
<span class="r-in"><span><span class="va">L</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         Contrasts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Levels   Batch1_vs_2 Batch3_vs_4 Batch1_vs_34</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Batch1           1           0          1.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Batch2          -1           0          0.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Batch3           0           1         -0.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Batch4           0          -1         -0.5</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Plot to visualize contrasts matrix</span></span></span>
<span class="r-in"><span><span class="fu"><a href="plotContrasts.html">plotContrasts</a></span><span class="op">(</span><span class="va">L</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="makeContrastsDream-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Fit linear mixed model for each gene</span></span></span>
<span class="r-in"><span><span class="co"># run on just 10 genes for time</span></span></span>
<span class="r-in"><span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="dream-method.html">dream</a></span><span class="op">(</span><span class="va">geneExpr</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>, <span class="va">form</span>, <span class="va">info</span>, L <span class="op">=</span> <span class="va">L</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># examine contrasts after fitting</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/coef.html" class="external-link">coef</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       Batch1_vs_2 Batch3_vs_4 Batch1_vs_34      Batch1      Batch2      Batch3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene1  0.46092425 -0.39590682    0.2556715 -10.1696108 -10.6305350 -10.6232357</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene2  0.49420745  0.01791921    0.3911706  -0.7408132  -1.2350206  -1.1230241</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene3 -0.08200976  0.32414781    0.3346561   0.3469564   0.4289662   0.1743742</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene4 -0.11531023 -0.01435608   -0.1300338  -4.6207874  -4.5054771  -4.4979316</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene5 -0.26795357 -0.26762214   -0.1398719   4.6243196   4.8922731   4.6303804</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene6  0.67049689  0.01251517    1.0410337  -2.4959563  -3.1664532  -3.5307325</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            Batch4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene1 -10.2273289</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene2  -1.1409433</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene3  -0.1497736</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene4  -4.4835755</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene5   4.8980025</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene6  -3.5432476</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># show results from first contrast</span></span></span>
<span class="r-in"><span><span class="fu"><a href="toptable-method.html">topTable</a></span><span class="op">(</span><span class="va">fit</span>, coef <span class="op">=</span> <span class="st">"Batch1_vs_2"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              logFC     AveExpr           t   P.Value adj.P.Val       z.std</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene6   0.67049689  -3.1554790  1.52366339 0.1319712 0.6497599  1.50637414</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene2   0.49420745  -1.1281610  1.36219015 0.1774115 0.6497599  1.34876797</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene1   0.46092425 -10.4664549  1.30835356 0.1949280 0.6497599  1.29613790</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene7   0.31606553  -4.3799381  0.84073462 0.4032967 0.8431315  0.83574794</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene9   0.35301723  -2.3079042  0.77799512 0.4391159 0.8431315  0.77368696</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene5  -0.26795357   4.7187640 -0.66862559 0.5058789 0.8431315 -0.66526828</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene10 -0.11579122  -2.3673775 -0.35847448 0.7210434 0.8940896 -0.35706470</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene4  -0.11531023  -4.5359748 -0.34244225 0.7330176 0.8940896 -0.34111426</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene3  -0.08200976   0.1702122 -0.24821747 0.8046806 0.8940896 -0.24729409</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene8  -0.02567263   0.9171386 -0.06725961 0.9465626 0.9465626 -0.06702401</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># show results from second contrast</span></span></span>
<span class="r-in"><span><span class="fu"><a href="toptable-method.html">topTable</a></span><span class="op">(</span><span class="va">fit</span>, coef <span class="op">=</span> <span class="st">"Batch3_vs_4"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              logFC     AveExpr           t    P.Value adj.P.Val       z.std</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene8  -0.80455799   0.9171386 -2.16904635 0.03340121 0.3340121 -2.12722716</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene1  -0.39590682 -10.4664549 -1.16045670 0.24967624 0.9767482 -1.15113612</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene3   0.32414781   0.1702122  1.01160906 0.31512596 0.9767482  1.00452431</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene10 -0.24451321  -2.3673775 -0.77854924 0.43881477 0.9767482 -0.77419620</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene5  -0.26762214   4.7187640 -0.68664313 0.49452493 0.9767482 -0.68312963</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene7   0.18507678  -4.3799381  0.50649399 0.61406680 0.9767482  0.50427691</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene9   0.02990159  -2.3079042  0.06772130 0.94619565 0.9767482  0.06748494</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene2   0.01791921  -1.1281610  0.05082868 0.95960356 0.9767482  0.05065108</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene4  -0.01435608  -4.5359748 -0.04385867 0.96513877 0.9767482 -0.04370599</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene6   0.01251517  -3.1554790  0.02924752 0.97674821 0.9767482  0.02914593</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># show results from third contrast</span></span></span>
<span class="r-in"><span><span class="fu"><a href="toptable-method.html">topTable</a></span><span class="op">(</span><span class="va">fit</span>, coef <span class="op">=</span> <span class="st">"Batch1_vs_34"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              logFC     AveExpr           t    P.Value adj.P.Val       z.std</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene6   1.04103369  -3.1554790  2.37923408 0.01998146 0.1998146  2.32669579</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene2   0.39117058  -1.1281610  1.08409908 0.28193902 0.9421247  1.07597371</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene3   0.33465610   0.1702122  1.01819011 0.31201734 0.9421247  1.01099809</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene1   0.25567153 -10.4664549  0.72973043 0.46791597 0.9421247  0.72587408</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene8  -0.27271522   0.9171386 -0.71831187 0.47489082 0.9421247 -0.71454406</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene4  -0.13003380  -4.5359748 -0.38839218 0.69886587 0.9421247 -0.38685188</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene5  -0.13987190   4.7187640 -0.35098159 0.72662357 0.9421247 -0.34962044</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene9   0.11152473  -2.3079042  0.24726976 0.80539615 0.9421247  0.24636959</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene7  -0.05691392  -4.3799381 -0.15221534 0.87944287 0.9421247 -0.15167551</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene10 -0.02340503  -2.3673775 -0.07285348 0.94212472 0.9421247 -0.07259963</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="http://gabrielhoffman.github.io" class="external-link">Gabriel Hoffman</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer></div>

  

  

  </body></html>

